tests/regression/test_line_smoother_periodic.py

```diff
@@ -0,0 +1,98 @@
+        new_coords[j, 0] = if(new_coords[j, 0] == 0 and Y < 0, 1, new_coords[j, 0])
+        new_coords[j, 0] = new_coords[j, 0] * Lx[0]
+        new_coords[j, 1] = old_coords[j, 2] * Ly[0]
+    end
+    """
+    cLx = Constant(1)
+    cLy = Constant(1)
+    par_loop((domain, instructions), dx,
+             {"new_coords": (new_coordinates, WRITE),
+              "old_coords": (old_coordinates, READ),
+              "Lx": (cLx, READ),
+              "Ly": (cLy, READ)},
+             is_loopy_kernel=True)
+    return Mesh(new_coordinates)
+
+
+def test_line_smoother_periodic():
+    N = 10
+    H = 0.1
+    nsmooth = 3
+    nref = 1
+
+    # Making a periodic hierarchy:
+    # first make a hierarchy of the cylinder meshes,
+    # then periodise each one
+    baseMesh = CylinderMesh(N, N, 1.0, H)
+
+    mh = MeshHierarchy(baseMesh, nref)
+    meshes = tuple(periodise(m) for m in mh)
+    mh = HierarchyBase(meshes, mh.coarse_to_fine_cells, mh.fine_to_coarse_cells)
+    mesh = mh[-1]
+
+    V = FunctionSpace(mesh, "CG", 1)
+    u = Function(V)
+    v = TestFunction(V)
+    bc = DirichletBC(V, Constant(0), [1, 2])
+
+    F = dot(grad(u), grad(v))*dx - inner(Constant(1), v)*dx
+
+    base = {"snes_type": "ksponly",
+            "ksp_type": "fgmres",
+            "ksp_monitor": None,
+            "ksp_max_it": 100,
+            "ksp_rtol": 1.0e-5}
+
+    mg_levels = {"ksp_max_it": nsmooth,
+                 "ksp_monitor": None,
+                 "ksp_norm_type": "unpreconditioned",
+                 "pc_type": "python",
+                 "pc_python_type": "firedrake.PatchPC",
+                 "patch_pc_patch_save_operators": True,
+                 "patch_pc_patch_local_type": "additive",
+                 "patch_pc_patch_construct_type": "python",
+                 "patch_pc_patch_construct_python_type": "firedrake.PlaneSmoother",
+                 "patch_pc_patch_construct_ps_sweeps": "0-%d" % (N+1),
+                 "patch_sub_ksp_type": "preonly",
+                 "patch_sub_pc_type": "lu"}
+
+    params = {"ksp_type": "richardson",
+              "ksp_richardson_self_scale": False,
+              "ksp_norm_type": "unpreconditioned",
+              "pc_type": "mg",
+              "pc_mg_type": "full",
+              "pc_mg_log": None,
+              "mg_levels": mg_levels,
+              "mg_coarse_pc_type": "python",
+              "mg_coarse_pc_python_type": "firedrake.AssembledPC",
+              "mg_coarse_assembled": {"mat_type": "aij",
+                                      "pc_type": "telescope",
+                                      "pc_telescope_subcomm_type": "contiguous",
+                                      "telescope_pc_type": "lu",
+                                      "telescope_pc_factor_mat_solver_type": "superlu_dist"}}
+
+    params = {**base, **params}
+
+    problem = NonlinearVariationalProblem(F, u, bcs=bc)
+    solver = NonlinearVariationalSolver(problem, solver_parameters=params)
+    solver.solve()
```

Comment:
Do you need the multigrid aspect to test the plane smoother, or is it sufficient to do a 1-level method. We're trying to reduce the total testing loading. If you need the multilevel to check some other things that's fine.

Reply:
I have reduced the size of the mesh (coarse grid is 3x3, fine grid is 6x6). Most of the runtime is in importing firedrake (at least on my machine).